### ISSUE TITLE
Failing tests when using Vagrant

### ISSUE THREAD

**Issue by techtonik** [gratipay/gratipay.com#3165]
`master` is currently failing for Vagrant.

```
=================================== FAILURES ===================================
________________________ TestPages.test_anon_can_browse ________________________

self = <test_pages.TestPages testMethod=test_anon_can_browse>

    def test_anon_can_browse(self):
>       self.browse()

tests/py/test_pages.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/py/test_pages.py:24: in browse
    r = self.client.GET(url, **kw)
env/lib/python2.7/site-packages/aspen/testing/client.py:81: in GET
    def GET(self, *a, **kw):    return self.hit('GET', *a, **kw)
env/lib/python2.7/site-packages/aspen/testing/client.py:105: in hit
    , return_after=return_after
env/lib/python2.7/site-packages/aspen/website.py:63: in respond
    , _return_after=return_after
env/lib/python2.7/site-packages/algorithm.py:288: in run
    new_state = function(**deps.as_kwargs)
env/lib/python2.7/site-packages/aspen/algorithms/website.py:95: in get_resource_for_request
    return {'resource': resources.get(website, dispatch_result.match)}
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

website = <aspen.website.Website object at 0x2b8d835cefd0>
fspath = '/vagrant/www/about/me/%redirect_to.spt'

    def get(website, fspath):
        """Given a website and a filesystem path, return a Resource object (with caching).
        """
    
        # XXX This is not thread-safe. It used to be, but then I simplified it
        # when I switched to diesel. Now that we have multiple engines, some of
        # which are threaded, we need to make this thread-safe again.
    
        # Get a cache Entry object.
        # =========================
    
        if fspath not in __cache__:
            entry = Entry()
            __cache__[fspath] = entry
    
        entry = __cache__[fspath]
    
    
        # Process the resource.
        # =====================
    
        mtime = os.stat(fspath)[stat.ST_MTIME]
        if entry.mtime == mtime:  # cache hit
            if entry.exc is not None:
                raise entry.exc
        else:  # cache miss
            try:
                entry.resource = load(website, fspath, mtime)
            except:  # capture any Exception
                entry.exc = (LoadError(traceback.format_exc()), sys.exc_info()[2])
            else:  # reset any previous Exception
                entry.exc = None
    
            entry.mtime = mtime
            if entry.exc is not None:
>               raise entry.exc[0]  # TODO Why [0] here, and not above?
E               LoadError: Traceback (most recent call last):
E                 File "/vagrant/env/lib/python2.7/site-packages/aspen/resources/__init__.py", line 168, in get
E                   entry.resource = load(website, fspath, mtime)
E                 File "/vagrant/env/lib/python2.7/site-packages/aspen/resources/__init__.py", line 137, in load
E                   resource = Class(website, fspath, raw, media_type, is_bound, mtime)
E                 File "/vagrant/env/lib/python2.7/site-packages/aspen/resources/simplate.py", line 46, in __init__
E                   pages = self.parse_into_pages(self.raw, self.is_bound)
E                 File "/vagrant/env/lib/python2.7/site-packages/aspen/resources/simplate.py", line 119, in parse_into_pages
E                   raise SyntaxError(msg)
E               SyntaxError:  resources must have at least three pages; /vagrant/www/about/me/%redirect_to.spt has one.

env/lib/python2.7/site-packages/aspen/resources/__init__.py:176: LoadError
```

TestPages.test_new_participant_can_browse fails with the same error.

**Comment by Changaco** [comment]
I'm guessing it's because they're symbolic links.

**Comment by techtonik** [comment]
Yes. Looks like symlinking is the cause.

**Comment by techtonik** [comment]
* [ ] www/about/me/%redirect_to.spt looks like this on Windows checkout:
```
../me.spt
```

**Comment by techtonik** [comment]
I am blocked. Why these symlinks exist? Should they be just removed? Should they emit a redirect? Or should they provide the information by including other template - gratipay/simplates.org#4 ?

**Comment by whit537** [comment]
The two symlinks at `www/about/me/{%redirect_to,index}.spt` exist in order to route all of these URLs to the same simplate (see #2791 for history):

https://gratipay.com/about/me
https://gratipay.com/about/me/
https://gratipay.com/about/me/foo/bar

They should not just be removed, no. We should either remove the symlinks by reimplementing the functionality in the `me.spt` simplate (maybe with an algorithm function for the redirect, and a `template/me.html` for the sign-in?), or you should [work around git symlinks on Windows](http://stackoverflow.com/questions/5917249/git-symlinks-in-windows).

**Comment by techtonik** [comment]
Why the three URLs with the same info?

**Comment by whit537** [comment]
@techtonik Technical limitations. See discussion on #2791.

**Comment by techtonik** [comment]
From http://aspen.io/virtual-paths/ it looks like it is possible to use:

  www/about/me/index.spt  - canonical template
  www/about/me                 - redirect to canonical
  www/about/me/%name    - flag (directory) to catch subpaths URL

**Comment by whit537** [comment]
@techtonik Ping @pjz here or on IRC for help with Aspen usage.

**Comment by techtonik** [comment]
@pjz, is it possible to get rid of symlinks in this scenario?

**Comment by pjz** [comment]
The full behavior for all possibilities (that we could think of) is documented in https://github.com/gratipay/aspen-python/blob/master/tests/dispatch_table_data.rst . The upshot is that latest behavior as of the last dispatcher rewrite is that a request for `www/about/me` will automatically redirect to `www/about/me/` _if there's no `me.spt` file in `about/`_.  Also, a request for `www/about/me/` in the presence of a `www/about/me/%sub.spt` file _will_  cause `%sub.spt` to be used, with `sub == ""`. 

So you should be able to nuke the symlinks and move `www/about/me.spt` to `www/about/me/%sub.spt`.

**Comment by techtonik** [comment]
That's great. I will need some time to get back to this. The table looks like it should belong to official docs.

**Comment by pjz** [comment]
It's meant to be, once aspen switches over to sphinx for docs.

**Comment by techtonik** [comment]
https://gratipay.com/about/me
"sign in please" page displayed
pushing bitbucket sign in gives redirect to https://gratipay.com/on/bitbucket/redirect
which redirects to https://gratipay.com/on/bitbucket/associate?oauth_verifier=629...&oauth_token=e...
which redirects to https://gratipay.com/about/me
which redirects to https://gratipay.com/techtonik/

**Comment by techtonik** [comment]
- [ ] remove `www/about/me.spt`
  - [ ] check that https://gratipay.com/about/me redirects to https://gratipay.com/about/me/
- [ ] replace everything in `www/about/me/` with `www/about/%sub.spt`
  - [ ] check that https://gratipay.com/about/me/history redirects to https://gratipay.com/techtonik/history

**Comment by techtonik** [comment]
Thanks everybody for info. It is now ready to be fixed in https://github.com/gratipay/gratipay.com/pull/3363 to finally get a working environment on Windows.

**Comment by techtonik** [comment]
@pjz I thought that `%sub` is some predefined name. Now I see..

**Comment by techtonik** [comment]
Python tests are fixed by #3318. JS tests are still failing, but are tracked in #3000. Closing. =)
